### PR TITLE
Vanilla-extract support

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

- Support vanilla-extract in Next config and storybook
- Demo of using design tokens and responsive styles
- Example of extending vanilla-based component with emotion

## Justify why they are needed

First steps towards new CSS-in-JS solution

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

File: packages/ui/src/components/Dialog/Dialog.tsx [hunk 1]
```diff
@@ -55,34 +16,39 @@ export const Content = (props: ContentProps) => {
...
 export const Trigger = DialogPrimitive.Trigger
 export const Close = DialogPrimitive.Close
+
+// Example of exposing base component for further styling by either vanilla-extract or emotion
+export const Window = ({
+  className,
+  ...forwardedProps
+}: { className?: string } & PropsWithChildren) => (
+  <div className={clsx(dialogWindow, className)} {...forwardedProps} />
+)
```
Comment (guilhermespopolin) [inline packages/ui/src/components/Dialog/Dialog.tsx:54]: 👍🏻

File: packages/ui/src/components/Dialog/Dialog.css.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      [animationAllowed]: {
+        animation: `${overlayShow} 500ms cubic-bezier(0.16, 1, 0.3, 1) forwards`,
+      },
+    },
+  },
+  frosted: {
+    backgroundColor: theme.colors.backgroundFrostedGlass,
+    backdropFilter: 'blur(64px)',
+  },
+})
```
Comment (guilhermespopolin) [inline packages/ui/src/components/Dialog/Dialog.css.ts:24]: [Discussion]
What do you think about the other notation where we add `@media` rules at the same level of all styles? Like we do for emotion styled components today. Should we set a common preference?

```
// theme
export const animationAllowed = '@media (prefers-reduced-motion: no-preference)'
```

```suggestion
export const overlay = styleVariants({
  base: {
    backgroundColor: 'rgba(0, 0, 0, 0.15)',
    position: 'fixed',
    inset: 0,
    [animationAllowed]: {
      animation: `${overlayShow} 500ms cubic-bezier(0.16, 1, 0.3, 1) forwards`,
    },
  },
  frosted: {
    backgroundColor: theme.colors.backgroundFrostedGlass,
    backdropFilter: 'blur(64px)',
  },
})
```

File: apps/store/src/components/ProductCard/ProductCard.tsx [hunk 3]
```diff
@@ -4,7 +4,7 @@ import Link from 'next/link'
 import { useRouter } from 'next/router'
 import { useTranslation } from 'next-i18next'
 import { useState } from 'react'
-import { Button, Space, theme, mq } from 'ui'
```
Comment (guilhermespopolin) [inline apps/store/src/components/ProductCard/ProductCard.tsx:7]: I guess I didn't follow this change? 🤔

**Review by guilhermespopolin — APPROVED** [review]
Nice work with this 💪🏻 .
It feels nice seeing regular HTML tags again. Looking forward to starting working with it.

File: packages/ui/src/components/Dialog/Dialog.css.ts (hunk 2 above)
Comment (alebedev) [reply packages/ui/src/components/Dialog/Dialog.css.ts:24]: Started with that, decided against it for 2 reasons
- type safety.  Not compatible with vanilla types unless you specifically downcast media rules to generic `string` which makes code more awkward
- compilation results are different.  Both variants work in Chrome, but I would not be 100% sure about all browsers

**Comment by alebedev** [comment]
### Merge activity

* **Feb 8, 9:54 AM EST**: @alebedev started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3879).

**Pull request merged**